server/src/test/scala/za/co/absa/atum/server/api/controller/PartitioningControllerUnitTests.scala

```diff
@@ -99,6 +104,20 @@ object PartitioningControllerUnitTests extends ZIOSpecDefault with TestData {
             failsWithA[InternalServerErrorResponse]
           )
         }
+      ),
+      suite("GetPartitioningSuite")(
```

Comment:
I would expect three test scenarios. 1. success case (you have that one OK). 2. Case where partitioning was not found, therefore the mock is set up to return NotFoundServiceError and you assert on NotFoundErrorResponse. 3. the unexpected error (you have that one already). You see with the mocks you are supposed to simulate all different scenarios.